File: src/lib/deepseekClient.ts
export class DeepSeekClient {
  constructor(private apiKey: string, private endpoint = "https://api.deepseek.com/v1") {}

  async generate(prompt: string) {
    const res = await fetch(`${this.endpoint}/generate`, {
      method: "POST",
      headers: {
        Authorization: `Bearer ${this.apiKey}`,
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ model: "DeepSeek-R1", prompt }),
    });
    if (!res.ok) {
      throw new Error(`DeepSeek error ${res.status}`);
    }
    return res.json();
  }
}

File: src-tauri/src/deepseek_client.rs
use reqwest::Client;
use serde::Serialize;

#[derive(Serialize)]
struct RequestBody<'a> {
    model: &'a str,
    prompt: &'a str,
}

pub async fn call_deepseek(
    api_key: &str,
    prompt: &str,
    endpoint: &str,
) -> Result<String, String> {
    let client = Client::new();
    let resp = client
        .post(format!("{}/generate", endpoint))
        .header("Authorization", format!("Bearer {}", api_key))
        .json(&RequestBody {
            model: "DeepSeek-R1",
            prompt,
        })
        .send()
        .await
        .map_err(|e| e.to_string())?;

    resp.text().await.map_err(|e| e.to_string())
}
